Data source changes for ECB, IMF

Detected during review of #105 (see e.g. [Travis job #200.1](https://travis-ci.org/dr-leo/pandaSDMX/jobs/577799702)) one test is failing for each of ECB and IMF.

The number of keys available for the ECB EXR dataflow has increased:
```
__________________________ test_request_preview_data ___________________________
    @pytest.mark.remote_data
    def test_request_preview_data():
        req = sdmx.Request('ECB')
    
        # List of keys can be retrieved
        keys = req.preview_data('EXR')
        assert isinstance(keys, list)
    
        # Count of keys can be determined
>       assert len(keys) == 4331
E       assert 4355 == 4331
E         -4355
E         +4331
tests/test_api.py:118: AssertionError
```

The IMF no longer provides the 'provisionagreement' endpoint. This can be confirmed via their web interface: go to https://sdmxcentral.imf.org/webservice/schema.html and try to select "Provision Agreement" from the "Structure Type" dropdown; the "Query Url:" box shows "N/A".
```
__________________ TestIMF.test_endpoints[provisionagreement] __________________
self = <tests.test_sources.TestIMF object at 0x7fe1ce242e10>
req = <pandasdmx.api.Request object at 0x7fe1ce242748>
endpoint = <Resource.provisionagreement: 'provisionagreement'>
    @pytest.mark.remote_data
    def test_endpoints(self, req, endpoint):
        # See pytest_generate_tests() for values of 'endpoint'
>       req.get(endpoint, tofile=self._cache_path.with_suffix('.' + endpoint))
tests/test_sources.py:107: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandasdmx/api.py:381: in get
    response.raise_for_status()

[snip]

E           requests.exceptions.HTTPError: 404 Client Error:  for url: https://sdmxcentral.imf.org/ws/public/sdmxapi/rest/provisionagreement/IMF/latest

../../../virtualenv/python3.7.1/lib/python3.7/site-packages/requests/models.py:940: HTTPError
```

The fixes are, respectively:
- Adjust the return value, and
- Xfail the test.

Also note that this issue uses a new GitHub label [`datasource`](https://github.com/dr-leo/pandaSDMX/labels/datasource). We can apply this to issues that are not caused by any breakage of pandaSDMX *per se*, but by changes in the behaviour of the different data sources.

--- a/pandasdmx/sources.json
+++ b/pandasdmx/sources.json
@@ -43,7 +43,10 @@
   {
     "id": "IMF",
     "url": "https://sdmxcentral.imf.org/ws/public/sdmxapi/rest",
-    "name": "International Monetary Fund"
+    "name": "International Monetary Fund",
+    "supports": {
+      "provisionagreement": false
+    }
   },
   {
     "id": "INEGI",

--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -115,7 +115,7 @@ def test_request_preview_data():
     assert isinstance(keys, list)
 
     # Count of keys can be determined
-    assert len(keys) == 4331
+    assert len(keys) == 4355
 
     # A filter can be provided, resulting in fewer keys
     keys = req.preview_data('EXR', {'CURRENCY': 'CAD+CHF+CNY'})

